back/engines/commercial/analysis/app/models/analysis/analysis.rb

```diff
@@ -41,7 +41,7 @@ class Analysis < ApplicationRecord
     def inputs
       scope = Idea.published
       if phase_id
-        scope.in_phase(phase_id)
+        scope.where(creation_phase_id: phase_id)
```

Comment:
This was the main issue, and this seems a better, more strict definition. phase_id is only set in case of a survey participation method, and this guarantees the associated custom_form is the right one.